common/src/main/java/net/neoforged/gradle/common/extensions/JavaVersionManager.java

```diff
@@ -0,0 +1,29 @@
+package net.neoforged.gradle.common.extensions;
+
+import org.gradle.api.Project;
+import org.gradle.api.plugins.JavaPluginExtension;
+import org.gradle.jvm.toolchain.JavaLanguageVersion;
+import org.gradle.jvm.toolchain.internal.JavaToolchain;
+
+import javax.inject.Inject;
+
+public class JavaVersionManager {
+
+    private final Project project;
+    private int javaVersion = -1;
+
+    @Inject
+    public JavaVersionManager(Project project) {
+        this.project = project;
+    }
+
+    public void setJavaVersion(int javaVersion, String context) {
+        if (this.javaVersion > javaVersion) {
+            project.getLogger().warn("Can not reconfigure java version from {} to {} for {}", this.javaVersion, javaVersion, context);
+            return;
+        }
+
+        this.javaVersion = javaVersion;
+        project.getExtensions().getByType(JavaPluginExtension.class).getToolchain().getLanguageVersion().set(JavaLanguageVersion.of(javaVersion));
```

Comment:
Gotta agree with maty here, this seems rather pointless...
...but if you are going to use this, you should at least make this `convention` not `set` so that user-added java versions override this one. This shouldn't override what users set, it should just provide something so that they don't _have_ to set it.